common/src/logging.rs

```diff
@@ -100,8 +100,46 @@ macro_rules! log_if_error_fmt {
     }};
 }
 
+pub struct Hidden<T> {
+    inner: T,
+}
+
+impl<T> Hidden<T> {
+    pub fn new(inner: T) -> Self {
+        Self { inner }
+    }
+}
+
+impl<T> fmt::Debug for Hidden<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "[FILTERED]")
+    }
+}
+
+impl<T> fmt::Display for Hidden<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "[FILTERED]")
+    }
+}
+
+impl<T> std::ops::Deref for Hidden<T> {
```

Comment:
How about to use [`derive_more`](https://crates.io/crates/derive_more) crate?
That'll make the implementation more compact: 

```rust
use derive_more::{From, Deref, DerefMut};

#[derive(Deref, DerefMut, From)]
pub struct Hidden(...);
```

Reply:
I'm not keen on another crate, we end up having to audit all of these crates